README.md

```diff
@@ -458,6 +458,27 @@ class MyApi::Base < JsonApiClient::Resource
 end
 ```
 
+
+### Type Casting
+
+You can define your own types and its casting mechanism for schema.
+
+```ruby
+require 'money'
+class MyMoneyCaster
+  def self.cast(value, default)
+    begin
+      Money.new(1000, "USD")
+    rescue ArgumentError
+      default
+    end
+  end
+end
+   
+JsonApiClient::Schema::TypeFactory.register money: MyMoneyCaster
```

Comment:
Small change: having the users register on the `TypeFactory` class locks to this implementation. It'll most likely remain this way, but it might make more sense for the developer to call `.register` on the `Schema` class/module instead.  If for whatever reason, we want to rename the `TypeFactory` class or refactor it, we need to keep that constant around. Internally, `JsonApiClient::Schema.register` could just delegate to `JsonApiclient::Schema::TypeFactory`